include/onnxruntime/core/graph/graph.h

```diff
@@ -147,19 +147,24 @@ class Node {
   Gets the function body if applicable otherwise nullptr
   @param try_init_func_body If not already intialized, initialize the function body
   (only applicable to operators which are defined as function in ONNX spec).
-  Function body can be initialized in 2 cases :
+  Function body can be initialized in 3 cases :
   1. For nodes of type "Fused"
-  2. For nodes which are defined as functions in ONNX spec (example: DynamicQuantizeLinear)
+  2. For nodes which are defined as functions in the spec (example: DynamicQuantizeLinear)
+  3. For nodes which reference a model local function. These functions are defined in the model itself and
+  do not have any schema associated with them.
   For all other cases this will always return nullptr.
   Nodes of type "Fused" are created during partitioning and the function body
   initialization for such nodes also happens during node creation. Therefore,
-  initialization of function body will happen via this method only in case 2 mentioned above.
+  initialization of function body will happen via this method only in cases 2 and 3 mentioned above.
   */
   const Function* GetFunctionBody(bool try_init_func_body = true);
 
   /** Gets the function body if applicable otherwise nullptr. */
   const Function* GetFunctionBody() const noexcept { return func_body_; }
 
+  /** Gets the mutable function body if applicable otherwise nullptr. */
+  Function* GetMutableFunctionBody(bool try_init_func_body = true);
+
 #endif
```

Comment:
Can the non-const GetFunctionBody just return Function*? Seems like two methods doing exactly the same thing.

Reply:
Do you mean : Function* GetMutableFunctionBody();
Internally the const function calls non-const function. Since we have not moved to lazy function initialization yet ( meaning only initialize function if we dont have a specialized kernel for it) function body initialization always happens before GetFunctionBody is called so yes right now the change should work... but we may need a mutable body with the capability to initialize when we do lazy initialization

Reply:
Why do we need a non-const GetFunctionBody that returns a const Function if we have GetMutableFunctionBody? 

Aren't those two non-const methods with a try_init_func_body parameter and the only difference is whether they return a const or mutable Function? But as they are non-const methods I don't quite see the value of one returning a const Function.